Jaxb assignment

zhyldyzMaks opened the pull request:
In this PR:
- xml file for the Course class created.
- JAXB annotations added to model classes from which the xml file is created. 
- DateAdapter class created.
- Marshaling and UnMarshaling methods tested in main method.

zhyldyzMaks commented:
Something went wrong with xml file. The fixed version has been sent on the last commit.

jishaGithub commented on src/main/resources/course.xml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
The requirement was to create xml file for at least 5 classes from the hierarchy.I dont see five classes included in your xml file.

jishaGithub replied:
No associated xsd file is included.

jishaGithub commented on src/main/java/com/solvd/db/utils/XmlParser.java:
```diff
@@ -0,0 +1,2 @@
+package com.solvd.db.utils;public class XmlParser {
```
XmlParser is empty

zhyldyzMaks commented:
Hey Jisha, how are you? I received this email from you, but cannot find the
comments on github PR. There is actually no XmlParser class since this is
the JAXB assignment PR and I added only files and classes related to JAXB
parsing. XML and XSD files are in other branch and PR. I created an XML
file for 5 classes for XML Parsing assignment. Since there was no the same
requirement for JAXB from Sergey, I did only one class xml file for test
purposes. I used Course class because all needed fields like Date, List,
Complex objects are there in that class. If I got the task wrong please let
me know to fix it. Thank you for your valuable comments.

On Sat, Jun 17, 2023 at 5:08 PM jishaGithub ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In src/main/resources/course.xml
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140301>:
>
> > @@ -0,0 +1,22 @@
> +<?xml version="1.0" encoding="UTF-8"?>
>
> The requirement was to create xml file for at least 5 classes from the
> hierarchy.I dont see five classes included in your xml file.
> ------------------------------
>
> In src/main/java/com/solvd/db/utils/XmlParser.java
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140598>:
>
> > @@ -0,0 +1,2 @@
> +package com.solvd.db.utils;public class XmlParser {
>
> XmlParser is empty
> ------------------------------
>
> In src/main/resources/course.xml
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140350>:
>
> > @@ -0,0 +1,22 @@
> +<?xml version="1.0" encoding="UTF-8"?>
>
> No associated xsd file is included.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#pullrequestreview-1484908245>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WZDN6COU76XIHW3GVHH33XLYTHJANCNFSM6AAAAAAZKBMN3M>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jishaGithub commented:
> Hey Jisha, how are you? I received this email from you, but cannot find the comments on github PR. There is actually no XmlParser class since this is the JAXB assignment PR and I added only files and classes related to JAXB parsing. XML and XSD files are in other branch and PR. I created an XML file for 5 classes for XML Parsing assignment. Since there was no the same requirement for JAXB from Sergey, I did only one class xml file for test purposes. I used Course class because all needed fields like Date, List, Complex objects are there in that class. If I got the task wrong please let me know to fix it. Thank you for your valuable comments.
> […](#)
> On Sat, Jun 17, 2023 at 5:08 PM jishaGithub ***@***.***> wrote: ***@***.**** requested changes on this pull request. ------------------------------ In src/main/resources/course.xml <[#3 (comment)](https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140301)>: > @@ -0,0 +1,22 @@ +<?xml version="1.0" encoding="UTF-8"?> The requirement was to create xml file for at least 5 classes from the hierarchy.I dont see five classes included in your xml file. ------------------------------ In src/main/java/com/solvd/db/utils/XmlParser.java <[#3 (comment)](https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140598)>: > @@ -0,0 +1,2 @@ +package com.solvd.db.utils;public class XmlParser { XmlParser is empty ------------------------------ In src/main/resources/course.xml <[#3 (comment)](https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140350)>: > @@ -0,0 +1,22 @@ +<?xml version="1.0" encoding="UTF-8"?> No associated xsd file is included. — Reply to this email directly, view it on GitHub <[#3 (review)](https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#pullrequestreview-1484908245)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A4WZDN6COU76XIHW3GVHH33XLYTHJANCNFSM6AAAAAAZKBMN3M> . You are receiving this because you authored the thread.Message ID: ***@***.***>

Hello Jyldyz,
I am doing good. Thanks for asking. I am sorry I deleted those comments , that's why you are not able to see it now. I was in a wrong PR , and I commented based on the first XML homework's requirements. I will be reviewing your first XML homework soon.

zhyldyzMaks commented:
I got it! Thank you so much

On Sat, 17 Jun 2023 at 21:17 jishaGithub ***@***.***> wrote:

> Hey Jisha, how are you? I received this email from you, but cannot find
> the comments on github PR. There is actually no XmlParser class since this
> is the JAXB assignment PR and I added only files and classes related to
> JAXB parsing. XML and XSD files are in other branch and PR. I created an
> XML file for 5 classes for XML Parsing assignment. Since there was no the
> same requirement for JAXB from Sergey, I did only one class xml file for
> test purposes. I used Course class because all needed fields like Date,
> List, Complex objects are there in that class. If I got the task wrong
> please let me know to fix it. Thank you for your valuable comments.
> … <#m_1825774820423624311_>
> On Sat, Jun 17, 2023 at 5:08 PM jishaGithub *@*.*> wrote: @.** requested
> changes on this pull request. ------------------------------ In
> src/main/resources/course.xml <#3 (comment)
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140301>>:
> > @@ -0,0 +1,22 @@ + The requirement was to create xml file for at least 5
> classes from the hierarchy.I dont see five classes included in your xml
> file. ------------------------------ In
> src/main/java/com/solvd/db/utils/XmlParser.java <#3 (comment)
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140598>>:
> > @@ -0,0 +1,2 @@ +package com.solvd.db.utils;public class XmlParser {
> XmlParser is empty ------------------------------ In
> src/main/resources/course.xml <#3 (comment)
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#discussion_r1233140350>>:
> > @@ -0,0 +1,22 @@ + No associated xsd file is included. — Reply to this
> email directly, view it on GitHub <#3 (review)
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#pullrequestreview-1484908245>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/A4WZDN6COU76XIHW3GVHH33XLYTHJANCNFSM6AAAAAAZKBMN3M
> . You are receiving this because you authored the thread.Message ID: *@*
> .***>
>
> Hello Jyldyz,
> I am doing good. Thanks for asking. I am sorry I deleted those comments ,
> that's why you are not able to see it now. I was in a wrong PR , and I
> commented based on the first XML homework's requirements. I will be
> reviewing your first XML homework soon.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zhyldyzMaks/Solvd_JDBC/pull/3#issuecomment-1595927464>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WZDN74YYLMGIG2JUOPZNDXLZQKPANCNFSM6AAAAAAZKBMN3M>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jishaGithub commented on src/main/java/com/solvd/db/utils/JaxbParser.java:
```diff
@@ -0,0 +1,62 @@
+    private static final Logger logger = LogManager.getLogger(JaxbParser.class);
+    private Course course;
+
+    public void unmarshalXml(){
```
```suggestion
    public void unmarshalXml() {
```

jishaGithub commented on src/main/java/com/solvd/db/mysql/model/Course.java:
```diff
@@ -0,0 +1,78 @@
+package com.solvd.db.mysql.model;
+
+import jakarta.xml.bind.annotation.*;
+
+import java.util.List;
```
```suggestion

import jakarta.xml.bind.annotation.*;
import java.util.List;

```

jishaGithub commented on src/main/java/com/solvd/db/mysql/model/Course.java:
```diff
@@ -0,0 +1,78 @@
+    @XmlElement(name = "exam")
+    private List<Exam> examList;
+
+    public Course(){}
```
```suggestion
    public Course() {}
```

jishaGithub commented on src/main/java/com/solvd/db/mysql/model/Exam.java:
```diff
@@ -0,0 +1,77 @@
+import jakarta.xml.bind.annotation.*;
+import jakarta.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+import java.sql.Date;
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "exam")
```
```suggestion
import java.sql.Date;

@XmlAccessorType(XmlAccessType.FIELD)
@XmlRootElement(name = "exam")
```
